Test/CoreSDK.Test/Shared/DataContracts/AvailabilityTelemetryTest.cs

```diff
@@ -0,0 +1,177 @@
+            telemetry.Properties.Add(new string('X', Property.MaxDictionaryNameLength) + 'Y', new string('X', Property.MaxValueLength + 1));
+
+            ((ITelemetry)telemetry).Sanitize();
+
+            Assert.Equal(new string('Z', Property.MaxAvailabilityMessageLength), telemetry.Message);
+            Assert.Equal(new string('Y', Property.MaxRunLocationLength), telemetry.RunLocation);
+            Assert.Equal(new string('D', Property.MaxTestNameLength), telemetry.TestName);
+
+            Assert.Equal(3, telemetry.Properties.Count); //AvailabilityTelemetry sanitize already sets one property which is why this is 3 instead of 2
+            Assert.Equal(new string('X', Property.MaxDictionaryNameLength), telemetry.Properties.Keys.ToArray()[0]);
+            Assert.Equal(new string('X', Property.MaxValueLength), telemetry.Properties.Values.ToArray()[0]);
+            Assert.Equal(new string('X', Property.MaxDictionaryNameLength - 3) + "001", telemetry.Properties.Keys.ToArray()[1]);
+            Assert.Equal(new string('X', Property.MaxValueLength), telemetry.Properties.Values.ToArray()[1]);
+
+            Assert.Same(telemetry.Properties, telemetry.Properties);
+        }
+
+        [TestMethod]
+        public void SanitizeWillAddDefaultFieldsForAvailability()
+        {
+            AvailabilityTelemetry telemetry = new AvailabilityTelemetry
+            {
+                Message = "Test Message",
+                RunLocation = "Test Location",
+                TestName = "Test Name",
+                TestTimeStamp = DateTimeOffset.Now,
+                Duration = TimeSpan.FromSeconds(30),
+                Success = true
+            };
+
+            ((ITelemetry)telemetry).Sanitize();
+
+            Assert.Equal(telemetry.Properties["FullTestResultAvailable"], "false");
+        }
+
+        [TestMethod]
+        public void SanitizeWillLeaveCustomMessageForAvailability()
+        {
+            AvailabilityTelemetry telemetry = new AvailabilityTelemetry
+            {
+                Message = "Test Message",
+                RunLocation = "Test Location",
+                TestName = "Test Name",
+                TestTimeStamp = DateTimeOffset.Now,
+                Duration = TimeSpan.FromSeconds(30),
+                Success = true
+            };
+
+            ((ITelemetry)telemetry).Sanitize();
+
+            Assert.Equal(telemetry.Data.message, "Test Message");
+        }
+
+        [TestMethod]
+        public void SanitizeWillSetFailedMessageForAvailability()
+        {
+            AvailabilityTelemetry telemetry = new AvailabilityTelemetry
+            {
+                RunLocation = "Test Location",
+                TestName = "Test Name",
+                TestTimeStamp = DateTimeOffset.Now,
+                Duration = TimeSpan.FromSeconds(30),
+                Success = false
+            };
```

Comment:
It might be worth considering turning these mostly identical object creation blocks in each test into a one liner helper method call to make the code more succinct.